src/wallet/wallet.cpp

```diff
@@ -3494,13 +3516,63 @@ bool CWallet::AddToWalletIfInvolvingMe(
     }
 }
 
-void CWallet::SyncTransaction(const CTransaction& tx, const CBlock* pblock, const int nHeight)
-{
-    LOCK(cs_wallet);
-    if (!AddToWalletIfInvolvingMe(Params().GetConsensus(), tx, pblock, nHeight, true))
-        return; // Not one of ours
+    void WalletBatchScanner::AddTransactionToBatch(const CTransaction &tx, const int nHeight)
+    {
+        decryptedNotes.insert(std::make_pair(
+            tx.GetHash(),
+            pwallet->TryDecryptShieldedOutputs(Params().GetConsensus(), tx, nHeight)));
```

Comment:
I did that elsewhere, but here specifically this is the interface between the node and its validation listeners, and none of the other listener APIs take the parameters.

I decided it would be both inconsistent for only one of the listener APIs to do so (hence I followed the existing pattern), and also inconsistent with future plans to split the wallet into a separate process: there, you'd set up / agree on the parameters when you register the listener, not in every individual API call.